server.js

```diff
@@ -82,13 +82,20 @@ function fetchPageTitle(muc, url) {
 }
 
 function sendBitcoinPrice(muc) {
-  fetch("https://api.coindesk.com/v1/bpi/currentprice/euro.json")
-    .then(res => res.json())
-    .then(json => {
-       const euro = json.bpi.EUR.rate_float;
-       const kollemate = Math.floor(euro / 1.5);
-       cl.sendRoomMessage(muc, `BTC: ${kollemate} x kolle-mate Flaschen / ${euro.toFixed(2)}€`);
-    });
+	var price;
+	fetch("http://matemat.hq.c3d2.de/summary.json")
+	.then(function(res){
+		return res.json();
+	}).then(function(json){
+		price = (json.filter(function(value){ return value.name=="kolle-mate";}))[0].price;
+	});
+	fetch("https://api.coindesk.com/v1/bpi/currentprice/euro.json")
+	.then(res => res.json())
+	.then(json => {
+		const euro = json.bpi.EUR.rate_float;
+		const kollemate = Math.floor(euro / price);
```

Comment:
Beachte daß in JavaScript (und vor allem in node.js) nichts blockiert (also den Kontrollfluß pausiert).

Während also das erste `fetch()` sich zum HTTP-Server verbindet, wird auch das zweite `fetch()` schon starten. Wenn Dinge parallel passieren, kann in 99% der Fälle deine Erwartung eintreffen, aber manchmal wird die zweite asynchrone Aktion (`fetch`) vor der ersten fertig. Dann wäre `price = undefined`.